### ISSUE TITLE
What is intended Swift 4 syntax?

### ISSUE THREAD

**Issue by swaffoja** [PromiseKit/OMGHTTPURLRQ#7]
I'm converting a project from Swift 2.3 to Swift 4. Old code was clean and simple using OMGHTTPURLRQ:

```
NSURLSession.GET("http://example.com", query: ["action" : "foo"])
```

When using the OMGHTTPURLRQ- extension I must cast every query parameter:

```
URLSession.shared.GET("http://example.com", query: ["action" as NSString : "foo" as NSString])
```

I'd rather not cast to NSObjects if possible. Would it make sense, and be appropriate, to change the method signatures to use `Dictionary<String, Any>` instead of `[NSObject: AnyObject]`. For example:

```
public func GET(_ url: String, query: Dictionary<String, Any>? = nil) -> URLDataPromise {
    return start(try OMGHTTPURLRQ.get(url, query) as URLRequest)
}
```

[See full example on GitHub](https://github.com/swaffoja/testomg/blob/master/testomg/ViewController.swift)

<img width="1560" alt="screen shot 2018-01-04 at 10 57 44 am" src="https://user-images.githubusercontent.com/887384/34574891-6e08fea8-f13e-11e7-9988-044dc3665c08.png">

**Comment by mxcl** [comment]
I'll take a look.

**Comment by mxcl** [comment]
I need to know what version of PromiseKit you are using, obviously.

**Comment by mxcl** [comment]
Fixed in ac0e2140d57c0193222582872586bf3ff30cc634. I will try to push to CocoaPods today.

**Comment by mxcl** [comment]
4.5.1 released.
